cloudinit/sources/DataSourceVMware.py

```diff
@@ -776,91 +774,52 @@ def get_default_ip_addrs():
     addresses associated with the device used by the default route for a given
     address.
     """
-    # TODO(promote and use netifaces in cloudinit.net* modules)
```

Comment:
Firstly, I suggest add a test/few tests for this get_default_ip_addrs() function in TestDataSourceVMware, It seems we only mock the return_value before, since this change focuses on this function, now test is needed.

Reply:
Thank you for the feedback!  Do you have any sense of the standard use cases this function gets called in?  I agree that the lack of unit tests makes it harder to validate my changes, but I would like to make sure any unit tests I write capture the original goals of the function rather than my perhaps imperfect understanding of what it should be.

Reply:
If possible, please mock return_value of netinfo.route_info() and netinfo.netdev_info() to verify below cases:
1. The interface used for the default route has only one ipv4, this function shall return the ipv4 address
2. The interface used for the default route has only one ipv6, this function shall return the ipv6 address
3. The interface used for the default route has both ipv4 and ipv6, this function shall return both ipv4 and ipv6 addresses
4. The interface used for the default route has multiple ipv4 and multiple ipv6,  this function shall return None, None
5. There is no default route, this function shall return None, None

Reply:
Hi @catmsred 
Overall the latest change looks good to me, only few nits.  
Please bear with me for one more week while I'm out of office for Chinese Spring Festival. In the meanwhile, I will check with @akutz if he want to take a look too.
Again, thanks for working on this.